code/obj/item/mops_cleaners.dm

```diff
@@ -1346,10 +1365,51 @@ TYPEINFO(/obj/item/handheld_vacuum/overcharged)
 			return
 		. = ..()
 
+	syndicate
+		name = "\improper GR1-Mk3 back tank"
+		desc = "A little label on the side reads \"Now with corrosive substances support!\"."
+		icon_state = "janitor_tank_s"
+		item_state = "janitor_tank_s"
+		initial_volume = 500
+		initial_reagents = list("sewage", "yuck", "badgrease", "oil", "toxic_slurry", "pizza", "space_drugs", "ants", "spiders", "miasma")
+		contraband = 4
+
+		New(loc, new_initial_reagents)
+			// Randomly fill up tank with junk
+			var/total_volume = src.initial_volume
+			var/list/reagent_pool = src.initial_reagents.Copy()
+			var/list/reagent_amounts = list()
+			while (total_volume > 0 && length(reagent_pool) > 0)
+				var/selected_reagent = pick(reagent_pool)
+				var/amt = 0
+				// Ensure the last reagent gets all remaining volume
+				if (length(reagent_pool) == 1)
+					amt = total_volume
+				else
+					// Dynamically adjust max and min to prevent underflow
+					var/max_amt = min(total_volume - (length(reagent_pool) - 1), max(10, total_volume / 5))
+					var/min_amt = min(10, max_amt)
+					amt = round(rand(min_amt, max_amt))
```

Comment:
I'm confused where some of these numbers are coming from- why divide total volume by 5? Why subtract the length of the reagent pool (I assume to get at least 1 unit of each reagent in?)
At the very least, this could use a more comprehensive comment to explain why this math is being done here.